hpcflow/tests/unit/test_task.py

```diff
@@ -1151,3 +1160,23 @@ def test_parameters_pending_during_add_task(workflow_w4, param_p2, param_p3):
     with workflow_w4.batch_update():
         workflow_w4.add_task(t2)
         assert workflow_w4._get_pending_add_parameter_keys()
+
+
+def test_add_task_before_and_after(workflow_w0):
+    t3 = Task(schemas=TaskSchema(objective="at_end", actions=[]))
+    t4 = Task(schemas=TaskSchema(objective="after_t1", actions=[]))
+    t5 = Task(schemas=TaskSchema(objective="at_start", actions=[]))
+    t6 = Task(schemas=TaskSchema(objective="before_t2", actions=[]))
+
+    with workflow_w0.batch_update():
+        workflow_w0.add_task_after(t3)
+        workflow_w0.add_task_after(t4, workflow_w0.tasks.t1)
+        workflow_w0.add_task_before(t5)
+        workflow_w0.add_task_before(t6, workflow_w0.tasks.t2)
+
+        assert workflow_w0.tasks[0].name == "at_start"
```

Comment:
I think we can only use one `assert` per test. I learnt this relatively recently so may have made the same mistake elsewhere in the code!

Reply:
Resolved in [c653f89](https://github.com/hpcflow/hpcflow-new/pull/348/commits/c653f8948112387f1a933ba05bc8be18ee64a9a1).